arches/app/models/models.py

```diff
@@ -1532,6 +1543,69 @@ def save(self, **kwargs):
         add_to_update_fields(kwargs, "graph_publication")
         super(ResourceInstance, self).save(**kwargs)
 
+        if should_update_resource_instance_lifecycle_state:
+            self.save_edit(
+                user=user,
+                edit_type="update_resource_instance_lifecycle_state",
+                oldvalue=f"{current_resource_instance_lifecycle_state.name} ({current_resource_instance_lifecycle_state.id})",
+                newvalue=f"{self.resource_instance_lifecycle_state.name} ({self.resource_instance_lifecycle_state.id})",
+                transaction_id=transaction_id,
+            )
+            self.run_lifecycle_handlers(
+                current_resource_instance_lifecycle_state,
+                request=request,
+                context=context,
+            )
+            if index:
+                self.index(context)
```

Comment:
I added `if index:` -- alternatively we can raise if we want index=False to be incompatible with updating lifecycle state.